packages/core/test/components/DataTable.test.tsx

```diff
@@ -483,47 +575,38 @@ describe('<DataTable /> handles edits', () => {
     expect(doneButton.find(Translate).prop('phrase')).toBe('Done');
   });
 
-  it('should enable edit mode and handle edit cancelation', () => {
-    // @ts-ignore
-    const wrapper = shallow(<DataTable {...props} instantEdit={false} />).dive();
-    const editButton = getHeader(wrapper)
-      .dive()
-      .dive()
-      .find(Button);
-    editButton.simulate('click');
-
-    const buttons = getHeader(wrapper)
-      .dive()
-      .dive()
-      .find(Button);
-    const cancelButton = buttons.at(0);
-    const applyButton = buttons.at(1);
-
-    cancelButton.simulate('click');
-    editButton.simulate('click');
-    applyButton.simulate('click');
-    editButton.simulate('click');
-
-    expect(wrapper.state('editMode')).toBe(true);
-    expect(cancelButton.find(Translate).prop('phrase')).toBe('Cancel');
-    expect(applyButton.find(Translate).prop('phrase')).toBe('Apply');
-  });
-
-  it('should replace text with inputs', () => {
+  it('should be editable', () => {
     // @ts-ignore
     const wrapper = mount(<DataTable {...props} />);
     const button = wrapper.find(TableHeader).find(Button);
     button.simulate('click');
     const grid = wrapper.find(Grid);
     const row = grid.find('[aria-rowindex=1]');
     const col = row.find('[aria-colindex=1]');
-    const input = col.find(Input);
-    expect(input.prop('value')).toBe(data[0].data.name);
+    const input = col.find(Input).find(FormInput);
+
+    const event = {
+      currentTarget: {
+        value: 'foo',
+      },
+    };
+
+    input.simulate('change', event);
+    input.simulate('click');
+
+    expect(
+      wrapper
+        .find(Grid)
+        .find('[aria-rowindex=1]')
+        .find('[aria-colindex=1]')
+        .find(Input)
+        .prop('value'),
+    ).toBe(data[0].data.name);
   });
 
-  it('should be editable', () => {
+  it('should be editable without instant edit', () => {
     // @ts-ignore
-    const wrapper = mount(<DataTable {...props} />);
+    const wrapper = mount(<DataTable {...props} instantEdit={false} />);
```

Comment:
not seeing it - but is `instantEdit` true in `props`?

Reply:
Yeah, `instantEdit` is default `true`. That happens in `DataTable/index` under `static defaultProps`

Reply:
aaah, k. i see it in the `DataTable/index` now used to seeing defaults where the props are defined / cause changes like that. with a default of false, usually, and true when the flag is set. defaultProps sometimes renders the falsey (undefined) value first before applying default prop values.